### PR TITLE
Add a splitPane option to duplicate current profile

### LINKED ISSUE #1756: Allow the user's "open pane" bindings to open a pane with the current pane
Currently, the split pane keybindings will always open the default profile. 

Perhaps a user will want to open a pane with the currently active pane's profile instead? 

This is _not_ #998, which is more about opening a pane with a specific profile, while this is about changing the default split behavior.

### PR DESCRIPTION
**Pull request by Surya-06**
This change adds the ability to configure a pane to split by duplicating the current profile 

Closes #1756

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1046,6 +1046,7 @@ namespace winrt::TerminalApp::implementation
     //   control which profile is created and with possible other
     //   configurations. See CascadiaSettings::BuildSettings for more details.
     void TerminalPage::_SplitPane(const TerminalApp::SplitState splitType,
+                                  const TerminalApp::SplitType splitMode,
```
[the remark this answers is not in the document]
Comment (Surya-06) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Added parameter comment to the existing list.

File: src/cascadia/TerminalApp/ActionArgs.h [hunk 2]
```diff
@@ -319,6 +319,7 @@ namespace winrt::TerminalApp::implementation
         SplitPaneArgs() = default;
         GETSET_PROPERTY(winrt::TerminalApp::SplitState, SplitStyle, winrt::TerminalApp::SplitState::None);
         GETSET_PROPERTY(winrt::TerminalApp::NewTerminalArgs, TerminalArgs, nullptr);
+        GETSET_PROPERTY(winrt::TerminalApp::SplitType, SplitMode, winrt::TerminalApp::SplitType::Manual);
 
         static constexpr std::string_view SplitKey{ "split" };
```
[the remark this answers is not in the document]
Comment (Surya-06) [reply src/cascadia/TerminalApp/ActionArgs.h]: I've added a key to extract the setting from the JSON but couldn't identify where it gets serialized under this key. Is this correct ? Can you please point me to the location where the serialization happens ( if needed ). Thanks !

[the remark this answers is not in the document]
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/ActionArgs.h]: Don't worry about it, we're not actually serializing keybindings at the moment 😄

**Comment by zadjii-msft** [comment]
@msftbot make sure @DHowett-MSFT signs off on this

**Comment by Surya-06** [comment]
@zadjii-msft Hey, sorry to disturb again but I'm still not able to deploy it to test my changes. The 'bcz' command gives me a couple of warnings but says that the build succeeded. When I try to deploy the solution though I run into different kinds of errors. Can you please help me out on testing this.
[The below error came up when I used deploy solution in VS]
image
[The same error comes up when I press f5 directly to debug too]
image

**Comment by zadjii-msft** [comment]
@Surya-06 I've frankly never seen that error before D: I'm not sure how that'd even happen... Does `E:\terminal\scr\cascadia\TerminalControl\Generated Files\SearchBoxControl.xaml` actually exist?

Maybe try just building the solution directly from VS, then running it there? There's pretty minor differences between how VS builds the solution and how plain-old MsBuild (which `bcz` calls) builds it. That's my only theory at the moment. 

I'll pull your branch locally to test myself and make sure.

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 3]
```diff
@@ -1064,7 +1066,23 @@ namespace winrt::TerminalApp::implementation
...
+        GUID realGuid;
+        bool profileFound = false;
+
+        if (splitMode == TerminalApp::SplitType::Duplicate)
+        {
+            std::optional<GUID> current_guid = focusedTab->GetFocusedProfile();
+            if (current_guid)
+            {
+                profileFound = true;
+                controlSettings = _settings->BuildSettings(current_guid.value());
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TerminalPage.cpp]:
```suggestion
                controlSettings = _settings->BuildSettings(current_guid.value());
                realGuid = current_guid.value();
```

**Comment by Surya-06** [comment]
@zadjii-msft I have committed your suggestion to handle the exception you pointed out. Let me know if I've missed anything.

**Comment by Surya-06** [comment]
@zadjii-msft Hey, I've finally been able to deploy and run the terminal. Is the key that I created enough to create a keyboard shortcut? If not is there a guide on how to create the shortcuts.

**Comment by zadjii-msft** [comment]
@Surya-06 yep, it works just fine for me :)

**Comment by Surya-06** [comment]
> @Surya-06 yep, it works just fine for me :)

can you tell me how you added the keyboard shortcut. i tried action as splitPane with splitMode as arg, i tried the value as 0/1 and also "duplicate" but couldn't see the result.

**Comment by zadjii-msft** [comment]
No problem, I just threw it on:

```json
        { "command": { "action": "splitPane", "split":"auto", "splitMode": "duplicate" }, "keys": [ "ctrl+alt+t" ] },
```

**Comment by Surya-06** [comment]
@zadjii-msft  Finally everything fell into place on my machine and I could use my changes. Thanks for all the help. Hope to contribute more!!!  😃 
PS: For anyone facing build errors, this video is super awesome. https://www.youtube.com/watch?v=4N1Ils4cDYo

**Comment by Surya-06** [comment]
@DHowett @carlos-zamora @miniksa @leonMSFT Can any of you please take a look at this, it's been a week since the last review. Thanks !

**Comment by miniksa** [comment]
Yeah, sure. Though I'm not sure why @zadjii-msft requested @DHowett-MSFT for sign-off. I'll confer with him about it.

**Comment by miniksa** [comment]
@msftbot make sure @DHowett-MSFT signs off on this

**Comment by Surya-06** [comment]
@miniksa Thanks for the quick reply. Do let me know if something needs to be done.

**Comment by DHowett-MSFT** [comment]
If your branch has maintainer access turned on, I'm going to push a commit that fixes the line endings making it difficult to review/making everything conflict w/ ActionAndArgs

**Comment by DHowett-MSFT** [comment]
(Sorry we sat on a relatively well-contained change for two weeks!)

**Comment by Surya-06** [comment]
> (Sorry we sat on a relatively well-contained change for two weeks!)

Thanks for responding so quick @DHowett-MSFT .

**Comment by say25** [comment]
Saw this in the release notes, but I can't seem to configure it in the `profiles.json`

**Comment by DHowett-MSFT** [comment]
@say25 okay we can only help you if you share with us what you've tried

**Comment by azchohfi** [comment]
You need to update to the latest version (0.10.761.0), then it works. Still, the schema is not giving me the splitMode property, on VSCode, but it works anyway. This is what I'm using:
```json
{ "command": { "action": "splitPane", "split": "horizontal", "splitMode": "duplicate"}, "keys": "alt+shift+-" },
{ "command": { "action": "splitPane", "split": "vertical", "splitMode": "duplicate"}, "keys": "alt+shift+plus" },
```

**Comment by say25** [comment]
Yea its more that it doesn't show in the schema but im trying:
```
    "keybindings": [
        {
            "command": {
                "action": "splitPane",
                "split": "horizontal",
                "splitMode": "manual"
            },
            "keys": [
                "ctrl+_"
            ]
        }, 
        .....
```

**Comment by DHowett-MSFT** [comment]
So, you almost certainly want the `duplicate` splitMode, not the `manual` one. The schema hasn't been updated yet (sorry about that!)

**Comment by say25** [comment]
Well I actually kinda have an "odd use case" basically I'll be at one directory and/or have some amount of env. variables set in the shell (AWS/Azure creds) and then I need to switch from bash to powershell or vice versa. Maybe this feature doesnt entirely address that but it looked like it might

**Pull request merged**